internal/pkg/manifest/svc.go

```diff
@@ -571,6 +576,121 @@ func sortExposedPorts(exposedPorts []ExposedPort) []ExposedPort {
 	return exposedPorts
 }
 
+// HTTPLoadBalancerTarget returns target container and target port for the ALB configuration.
+// This method should be called only when ALB config is not empty.
+func (s *LoadBalancedWebService) HTTPLoadBalancerTarget() (targetContainer string, targetPort string, err error) {
+	exposedPorts, err := s.ExposedPorts()
+	if err != nil {
+		return "", "", err
+	}
+	// Route load balancer traffic to main container by default.
+	targetContainer = aws.StringValue(s.Name)
+	targetPort = s.MainContainerPort()
+
+	rrTargetContainer := s.RoutingRule.TargetContainer
+	rrTargetPort := s.RoutingRule.TargetPort
+	if rrTargetContainer == nil && rrTargetPort == nil { // both targetPort and targetContainer are nil.
+		return
+	}
+
+	if rrTargetPort == nil { // when target_port is nil
+		if rrTargetContainer != s.Name {
+			targetContainer = aws.StringValue(rrTargetContainer)
+			targetPort = aws.StringValue(s.Sidecars[aws.StringValue(rrTargetContainer)].Port)
+		}
+		return
+	}
+
+	if rrTargetContainer == nil { // when target_container is nil
+		container, port := httpLoadBalancerTarget(exposedPorts, rrTargetPort)
+		targetPort = aws.StringValue(port)
+		if container != nil {
+			targetContainer = aws.StringValue(container)
+		}
```

Comment:
How should the caller interpret it when `targetContainer` is returned as `""` from `func (s *LoadBalancedWebService) HTTPLoadBalancerTarget()`?

Reply:
I don't think that would the case because at the beginning of the function we set `targetContainer = aws.StringValue(s.Name)` So main container name would be `targetContainer`. Does that make sense?